Misc bug fixes from the recent merges.

pop opened the pull request:
I was getting a few errors when trying to converge the cookbook and this PR should fix those errors.
1. Testing was broken between local and openstack instances of the cookbook. This was probably just an oversight of mine but I added separate test suites for local and openstack deployment.
2.  There were a few small errors in bootstrap_user.rb (DJANGO_SETTINGS_MODULE being incorrectly set was the largest one) so I fixed those.
3. There was a pathing issue in so I corrected that.
4. .kitchen.cloud.yml and .kitche.yml had some attributes which needed to be modified since we are placing everything in /opt instead of $HOME.

pop committed:
Misc bug fixes from the recent merges.
Added seperate test suites for local and openstack deployment.
Fixed a bootstrap_user.rb issue
Fixed a path issue in default.rb recipe.

armiller commented on test/integration/cloud/serverspec/pgd_spec.rb:
```diff
@@ -0,0 +1,34 @@
+
+describe package('libffi-devel') do
+  it { should be_installed }
+end
```
I would merge these two package tests.

``` Ruby
%w(python27 libffi-devel).each do |p|
  describe package(p) do
    it { should be_installed }
  end
end
```

jordane replied:
I agree. This is standard Ruby; we do it everywhere else.

armiller commented:
Other than that one nit LGTM.

mathuin commented:
So I checked out the branch and ran `tkc test` and it passed.  I then ran `kitchen test` and _it_ passed.  I'm ambivalent on @armiller's nit.  +1

jordane commented on test/integration/cloud/serverspec/pgd_spec.rb:
```diff
@@ -0,0 +1,34 @@
+  it { should be_directory }
+end
+
+describe file('/opt/pgd/selected_proteins.txt') do
```
You should combine this with the other `describe file` that `should be_file`, just like @armiller mentioned with the packages.

jordane commented on test/integration/cloud/serverspec/pgd_spec.rb:
```diff
@@ -0,0 +1,34 @@
+  it { should be_directory }
+end
+
+describe file('/opt/vrt') do
```
Combine this with the other `should be_directory` please.

jordane commented on test/integration/local/serverspec/pgd_spec.rb:
```diff
@@ -0,0 +1,34 @@
+  it { should be_installed }
+end
+
+describe package('libffi-devel') do
```
Same applies here.

jordane commented on test/integration/local/serverspec/pgd_spec.rb:
```diff
@@ -0,0 +1,34 @@
+end
+
+describe file('/opt/vrt') do
+  it { should be_directory }
```
and here.

jordane commented on test/integration/local/serverspec/pgd_spec.rb:
```diff
@@ -0,0 +1,34 @@
+end
+
+describe file('/opt/pgd/selected_proteins.txt') do
+  it { should be_file }
```
and here.

pop committed:
Syntax fixes and PR #20 adjustments

mathuin commented:
I reran both tests.  They both passed.  +1

jordane commented on test/integration/cloud/serverspec/pgd_spec.rb:
```diff
@@ -0,0 +1,26 @@
+require 'serverspec'
```
I noticed that this file is functionally equivalent to the others named `pgd_spec.rb` in other suites. It'd be nice if we could refactor so that only one copy of this file exists in any one place.

Could you try moving this file outside of `test/integration/cloud`, and then symlinking `test/integration/cloud/serverspec/pgd_spec.rb` to it? My hope is that busser will still copy it over and run the tests correctly, but I worry that it will not do so as I hope. If it did, we could easily refactor these down to one file.

If it doesn't work that way then I'm not sure how to easily refactor these three files. @armiller might have an idea.

pop committed:
pared down pgd_spec.rb to one file

mathuin committed:
Removed unused vagrant.json file

The vagrant.json file is no longer used.  I removed it and ran kitchen
test both locally in the cloud, and both sets of tests passed.

jordane committed:
Merge pull request #22 from osuosl-cookbooks/remove-vagrant-json

Removed unused vagrant.json file

armiller commented:
Whats up with this pr? Does it still need to be merged?

armiller commented:
ping again

mathuin commented:
In attempting to test this, I discovered that my kitchen configuration is apparently out of date.

This is what I get when I run `kitchen converge`:

```
-----> Starting Kitchen (v1.6.0)
$$$$$$ No platforms supported in metadata.rb;showing all available platforms
[2016-05-12T15:30:18-07:00] WARN: Ohai::Config[:log_level] is set. Ohai::Config[:log_level] is deprecated and will be removed in future releases of ohai. Use ohai.log_level in your configuration file to configure :log_level for ohai.
[2016-05-12T15:30:18-07:00] WARN: Ohai::Config[:log_location] is set. Ohai::Config[:log_location] is deprecated and will be removed in future releases of ohai. Use ohai.log_location in your configuration file to configure :log_location for ohai.
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::UserError
>>>>>> Message: Kitchen::Driver::Openstack<local-centos-67>#config[:private_key_path] cannot be blank
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

Is there something obvious I'm missing?

armiller commented:
Check to see if you env vars match this:
https://docs.osuosl.org/software/openstack/openstack_test_kitchen.html#environment-variables
and try again.

mathuin commented:
After some tweaking with my environment I was able to get the Openstack test kitchen stuff working.

The PGD code base has moved past the point where this PR can work.  I'm going to examine it in detail -- and the other outstanding PR on the similar topic -- to determine what if anything can be repurposed into a new PR.

ramereth commented:
@mathuin shall we just go ahead and close this then?

mathuin commented:
@ramereth Right now, I would say yes.  The codebase had a reorg and it's due for another once the scientific changes are approved.  Those architectural changes are significant enough that it will break the cookbook pretty hard.

I recommend we close this issue, and defer any cookbook-related issues until a plan is in place to migrate PGD off of zeus.  It's likely that by that time, we'll be on whatever comes after chef. :-)
